fsm/join_add_cds.go

```diff
@@ -124,106 +123,106 @@ func JoinAddCdsAction(z *music.Zone) bool {
+	cdsFromKSK := map[uint16]*dns.CDS{}     // cdsFromKSK: map of CDS RRs created from all KSKs from all signers
+	cdsFromSigners := map[uint16]*dns.CDS{} // cdsFromSigners: map of all CDS RRs found by querying all signers
 
-	cdnskeymap := map[uint16]*dns.CDNSKEY{}
-	cdnskeymap2 := map[uint16]*dns.CDNSKEY{}
+	cdnskeyFromKSK := map[uint16]*dns.CDNSKEY{}
+	cdnskeyFromSigners := map[uint16]*dns.CDNSKEY{}
 
-	for _, s := range z.SGroup.SignerMap {
-		updater := music.GetUpdater(s.Method)
-		log.Printf("VerifyCdsPublished: %s Using FetchRRset interface\n", z.Name)
-		err, rrs := updater.FetchRRset(s, z.Name, z.Name, dns.TypeDNSKEY)
+	// Fetch DNSKEYS
+	for _, signer := range zone.SGroup.SignerMap {
+		updater := music.GetUpdater(signer.Method)
+		log.Printf("VerifyCdsPublished: %s Using FetchRRset interface\n", zone.Name)
+		err, rrSet := updater.FetchRRset(signer, zone.Name, zone.Name, dns.TypeDNSKEY)
 		if err != nil {
 			log.Printf("Error from updater.FetchRRset: %v\n", err)
 		}
 
 		// Create CDS/CDNSKEY RRsets
-		for _, a := range rrs {
+		for _, a := range rrSet {
 			dnskey, ok := a.(*dns.DNSKEY)
 			if !ok {
 				continue
 			}
 
 			if f := dnskey.Flags & 0x101; f == 257 {
-				cdsmap[dnskey.KeyTag()] = dnskey.ToDS(dns.SHA256).ToCDS()
-				cdnskeymap[dnskey.KeyTag()] = dnskey.ToCDNSKEY()
+				cdsFromKSK[dnskey.KeyTag()] = dnskey.ToDS(dns.SHA256).ToCDS()
+				cdnskeyFromKSK[dnskey.KeyTag()] = dnskey.ToCDNSKEY()
 			}
 		}
 	}
-	keyids := []uint16{}
-	for k, _ := range cdsmap {
+	var keyids []uint16
+	for k, _ := range cdsFromKSK {
 		keyids = append(keyids, k)
 	}
 	log.Printf("Verify Publication of CDS: there are KSKs at the signers with the following keytags: %v\n", keyids)
 
-	// Check against published CDS/CDNSKEY RRsets.
-	for _, signer := range z.SGroup.SignerMap {
+	// Compare DNSKEYs to published CDS/CDNSKEY RRsets.
+	for _, signer := range zone.SGroup.SignerMap {
 		updater := music.GetUpdater(signer.Method)
-		err, cdsrrs := updater.FetchRRset(signer, z.Name, z.Name, dns.TypeCDS)
+		err, cdsRRset := updater.FetchRRset(signer, zone.Name, zone.Name, dns.TypeCDS)
 		if err != nil {
-			err, _ = z.SetStopReason(fmt.Sprintf("Unable to fetch CDS RRset from %s: %v",
+			err, _ = zone.SetStopReason(fmt.Sprintf("Unable to fetch CDS RRset from %s: %v",
 				signer.Name, err))
 			return false
 		}
-		err, cdnskeyrrs := updater.FetchRRset(signer, z.Name, z.Name, dns.TypeCDNSKEY)
+		err, cdnskeyRRset := updater.FetchRRset(signer, zone.Name, zone.Name, dns.TypeCDNSKEY)
 		if err != nil {
-			err, _ = z.SetStopReason(fmt.Sprintf("Unable to fetch CDNSKEY RRset from %s: %v",
+			err, _ = zone.SetStopReason(fmt.Sprintf("Unable to fetch CDNSKEY RRset from %s: %v",
 				signer.Name, err))
 			return false
 		}
 
-		for _, cdsrr := range cdsrrs { // check all CDS RRs from this signer
-			cds, ok := cdsrr.(*dns.CDS)
+		for _, rr := range cdsRRset {
+			cdsRR, ok := rr.(*dns.CDS)
```

Comment:
Here cdsRR is a *dns.CDS

Reply:
Ok, I see that. Do you mean that it is incorrect in some way? I don't have enough information in your comment with my knowledge level to make a determination if there is something wrong or not. If there is something wrong do you have a suggestion for an improvement?